### PR TITLE
feat: client support 'priority' parameter

### PR DESCRIPTION
**Pull request by bigerous**
Signed-off-by: bigerous <cuidajun.cdj@alibaba-inc.com>

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: cmd/dfget/cmd/root.go [hunk 1]
```diff
@@ -160,6 +160,8 @@ func init() {
 
 	flagSet.StringP("pattern", "p", dfgetConfig.Pattern, "The downloading pattern: p2p/seed-peer/source")
```
Comment (gaius-qi) [inline cmd/dfget/cmd/root.go:161]: Use `MarkDeprecated`.

File: pkg/net/http/http.go [hunk 2]
```diff
@@ -46,3 +49,17 @@ func PickHeader(header http.Header, key, defaultValue string) string {
 
 	return defaultValue
 }
+
+// PickTypedHeader pick typed header with key.
+func PickTypedHeader[T any](header http.Header, key string, defaultValue T) T {
```
Comment (gaius-qi) [inline pkg/net/http/http.go:54]: Refactor `PickHeader` and remove `PickTypedHeader`.

**Review by gaius-qi — CHANGES_REQUESTED** [review]

File: pkg/net/http/http.go (hunk 2 above)
Comment (bigerous) [reply pkg/net/http/http.go]: PickTypedHeader has been removed

File: client/daemon/transport/transport.go [hunk 3]
```diff
@@ -268,6 +280,12 @@
 	filter := nethttp.PickHeader(req.Header, config.HeaderDragonflyFilter, rt.defaultFilter)
 	tag := nethttp.PickHeader(req.Header, config.HeaderDragonflyTag, rt.defaultTag)
 	application := nethttp.PickHeader(req.Header, config.HeaderDragonflyApplication, rt.defaultApplication)
+	var priority = rt.defaultPriority
+	priorityString := nethttp.PickHeader(req.Header, config.HeaderDragonflyPriority, fmt.Sprintf("%d", rt.defaultPriority))
+	priorityInt, err := strconv.Atoi(priorityString)
+	if err != nil {
+		priority = commonv1.Priority(priorityInt)
```
[the remark this answers is not in the document]
Comment (bigerous) [reply client/daemon/transport/transport.go:287]: fixed

**Review by gaius-qi — APPROVED** [review]
LGTM

**Pull request merged**